src/lexer.cpp

```diff
@@ -0,0 +1,109 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "lexer.h"
+
+//VARIABLES
+std::vector<std::string> all_types = {"PROGRAM", "VAR", "END_VAR", "END_PROGRAM", "BOOL", "TRUE", "FALSE"}; //keywords
+std::vector<std::string> operators = {"<", ">", "+", "-", "%", ":="}; // operators
+std::vector<std::string> all_sc = {"{", "}", "[", "]", ";", ":"}; // special characters
+std::vector<Token> fin_tokens;
```

Comment:
Global variables shouldn't be used like this. I think it's fine to have constants be global, such as the vectors of identifiers, however the return result of the function shouldn't be global.